src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,111 @@
+import { useEffect, useState, useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
 
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { sortPeople } from '../utils/sortPeople';
+
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setLoading] = useState(false);
+  const [isError, setError] = useState(false);
+  const [searchParams] = useSearchParams();
+  const sex = searchParams.get('sex');
+  const query = searchParams.get('query');
+  const centuries = searchParams.getAll('centuries');
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  async function getPeopleList() {
+    try {
+      setLoading(true);
+
+      const peopleList = await getPeople();
+
+      setPeople(peopleList);
+    } catch (error) {
+      setError(true);
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  function filterByQuery(str: string | null) {
+    return str?.toLowerCase().includes(query?.toLowerCase() || '') || false;
+  }
+
+  const shownPeopleList = useMemo(() => {
+    let filteredArray = [...people];
+
```

Comment:
No need to create copy, cuz  filter return new Array `sortPeople` create new one